TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/Vuforia.java

```diff
@@ -32,35 +34,113 @@ public void close() {
+    private static final String VUFORIA_KEY = "Ad0Srbr/////AAABmdpa0/j2K0DPhXQjE2Hyum9QUQXZO8uAVCNpwlogfxiVmEaSuqHoTMWcV9nLlQpEnh5bwTlQG+T35Vir8IpdrSdk7TctIqH3QBuJFdHsx5hlcn74xa7AiQSJgUD/n7JJ2zJ/Er5Hc+b+r616Jf1YU6RO63Ajk5+TFB9N3a85NjMD6eDm+C6f14647ELnmGC03poSOeczbX7hZpIEObtYdVyKZ2NQ/26xDfSwwJuyMgUHwWY6nl6mk0GMnIGvu0/HoGNgyR5EkUQWyx9XlmxSrldY7BIEVkiKmracvD7W9hEGZ2nPied6DTY5RFNuFX07io6+I59/d7291NXKVMDnFAqSt4a2JYsECv+j7b25S0mD";
 
-    private static final float mmPerInch        = 25.4f;
+    private static final float mmPerInch        = 25.4f;                    // constant for converting measurements from inches to millimeters
     private static final float mmTargetHeight   = (6) * mmPerInch;          // the height of the center of the target image above the floor
-    private static final float stoneZ = 2.00f * mmPerInch;
-    private static final float bridgeZ = 6.42f * mmPerInch;
-    private static final float bridgeY = 23 * mmPerInch;
-    private static final float bridgeX = 5.18f * mmPerInch;
-    private static final float bridgeRotY = 59;                                 // Units are degrees
-    private static final float bridgeRotZ = 180;
-    private static final float halfField = 72 * mmPerInch;
+
+    private static final float halfField = 72 * mmPerInch;                  // constants for perimeter targets
     private static final float quadField  = 36 * mmPerInch;
-    private OpenGLMatrix lastLocation = null;
+
+    private OpenGLMatrix lastLocation = null; // class members
     private VuforiaLocalizer vuforia = null;
     private float phoneXRotate    = 0;
     private float phoneYRotate    = 0;
     private float phoneZRotate    = 0;
-    public VuforiaTrackables targetsSkyStone;
+    public VuforiaTrackables targetsUltGoal;
     private List<VuforiaTrackable> allTrackables;
 
-    public Vuforia(HardwareMap hardwareMap, CameraChoice choice) {
-        vuforia = setCamera(hardwareMap, choice);
+    public Vuforia(HardwareMap hardwareMap, CameraChoice cameraChoice) {
+        vuforia = initVuforia(hardwareMap, cameraChoice);
+
+        targetsUltGoal = vuforia.loadTrackablesFromAsset("UltimateGoal");
+
+        VuforiaTrackable blueTowerGoalTarget = targetsUltGoal.get(0);
+        blueTowerGoalTarget.setName("Blue Tower Goal");
+        VuforiaTrackable redTowerGoalTarget = targetsUltGoal.get(1);
+        redTowerGoalTarget.setName("Red Tower Goal");
+        VuforiaTrackable redAllianceTarget = targetsUltGoal.get(2);
+        redAllianceTarget.setName("Red Alliance");
+        VuforiaTrackable blueAllianceTarget = targetsUltGoal.get(3);
+        blueAllianceTarget.setName("Blue Alliance");
+        VuforiaTrackable frontWallTarget = targetsUltGoal.get(4);
+        frontWallTarget.setName("Front Wall");
+
+        // For convenience, gather together all the trackable objects in one easily-iterable collection */
```

Comment:
Good Data Structures question: is this necessary? Why _might_ we want to do this?